Added REST Controllers + service fixes

MisaHack opened the pull request:
Added : CharacterController, MovieController, RoleController, UserController, VoteController

MisaHack committed:
Added REST Controllers + service fixes

Added : CharacterController, MovieController, RoleController, UserController, VoteController

branimir-vujicic commented on src/main/java/com/springboot/star_wars_character_popularity/app/controller/CharacterController.java:
```diff
@@ -0,0 +1,50 @@
+        return characterService.getAllCharacters();
+    }
+
+    @GetMapping("{id}")
```
use /{id} on path definition

branimir-vujicic replied:
same for the rest of mappings

MisaHack replied:
Ok. "/{id}" added where it was needed.

branimir-vujicic reviewed (commented):
You should add one feature at a time along with the tests.
In this case: you should add a movie controller, service, and tests.

stefanbesovic commented on src/main/java/com/springboot/star_wars_character_popularity/app/controller/MovieController.java:
```diff
@@ -0,0 +1,50 @@
+
+    @Autowired
+    private MovieService movieService;
+
```
No need for @Autowire. 

Try using @RequiredArgsConstructor annotation(lombok) on top of the class and remove constructor and set field as final.

MisaHack committed:
Added CharacterServiceUnitTest + changes to CharacterModel

Added : CharacterServiceUnitTest, changes to CharacterModel

MisaHack committed:
Added verifies to Tests

Updated : CharacterServiceUnitTest, MovieServiceUnitTest, RoleServiceUnitTest, UserServiceUnitTest, VoteServiceUnitTest

stefanbesovic commented on src/main/java/com/springboot/star_wars_character_popularity/app/model/MovieModel.java:
```diff
@@ -21,6 +21,10 @@
 
+    public MovieModel(String name){
+        this.name = name;
+    }
```
I guess it's better to use @Builder annotation on top of the class and to use it when you are initializing object, to prevent this boilerplate code.

stefanbesovic commented on src/test/java/com/springboot/star_wars_character_popularity/app/service/CharacterServiceUnitTest.java:
```diff
@@ -90,21 +91,43 @@ void shouldReturnSavedCharacterById(){
+        CharacterModel actualModel = characterService.updateCharacter(characterModel, id);
+
+        //then
+        assertEquals("Jedi", actualModel.getName());
```
I think that this checks are unnecessary, because it's always gonna be true. (you hardcoded what you want that method to return)
Maybe it's better to use verify() method to check if the actual method is called.

MisaHack replied:
Ok. Added verify() methods.

stefanbesovic commented on src/test/java/com/springboot/star_wars_character_popularity/app/service/MovieServiceUnitTest.java:
```diff
@@ -0,0 +1,106 @@
+
+        //then
+        assertNotNull(actualModel);
+        assertEquals("A New Hope", actualModel.getName());
```
same here

MisaHack replied:
Ok. Added verify() methods.
